WordPress/gutenberg
# [RNMobile]: Fix search block focus issues

**NOTE: ~[This PR](https://github.com/WordPress/gutenberg/pull/29275) must be merged before this one can be taken out of draft~**

**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/3210

## Description

Fixes an couples issues found in a previous PR review:
- iOS: When adding a new search block when there are other text-based blocks already on the page would cause a bizarre UI issue where the search block kept stealing the focus causing a strange blinking of focusing between the search block and the other text blocks.
- Also on iOS when adding the search block, the app would appear to hang for a while before finally adding the block. 
- Clicking on the placeholder text input when the search block was not selected would not select the block. 

## How has this been tested?



Tested on both Android emulator and iOS simulator
1. Add a text block to the page. 
2. Add a search block. Notice the placeholder text input has the focus. 
3. Click the text in the text block added from step 1. Notice the focus remains on the text block as expected. 
4. Click the placeholder text. Notice the search block is now selected.
5. Click the button text of the search block and modify it. Verify successful.
6. Click the search label text and modify it. Verify successful. 

## Screenshots 

iOS before - search block stealing focus from text-based blocks

https://user-images.githubusercontent.com/5810477/109748272-96386d00-7ba6-11eb-9c0a-c8bc165318d6.mov

iOS After:

https://user-images.githubusercontent.com/5810477/109748324-ad775a80-7ba6-11eb-8e35-7306ddc8a732.mov

Android before - selecting placeholder text field when block not selected:

https://user-images.githubusercontent.com/5810477/109748446-e31c4380-7ba6-11eb-8fd4-cb7d010309c0.mov



Android after - placeholder text is focused when block first added:

https://user-images.githubusercontent.com/5810477/109748436-df88bc80-7ba6-11eb-9ff1-68e8e73e3b19.mov



## Types of changes




- Started tracking when different components of the `Search Block` are focused and passing that state via the `isSelected` property of those components. 
- Passed the `Search Block`'s `onFocus` prop to it's child components so when the placeholder text was directly clicked the block would become selected.
- Swapped out the `TextInput` component with the `PlainText` component. 
- Added a `blur()` method to the `PlainText` component and then added a `useEffect` to be called when the`isSelected` property changes to force the `PlainText` to blur when the `Search Block` is no longer selected. Similar things are done in other blocks (`Image`, `MediaText`, `Video`) except they implement a static `getDerivedStateFromProps` method to do the same thing. It works in those blocks because they are React _classes_. My `Search Block` is a _Function Component_ so I'm using `useEffect` to accomplish the same thing. 

## Checklist:
